packages/examples/file-uploads/src/views/file-uploader.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import { uploadFile } from '../actions/files'
+import { Subject } from 'microcosm'
+import { Presenter, ActionForm } from 'microcosm-dom'
+import { Progress } from './progress'
+
+function asFormData(form) {
+  return { data: new FormData(form) }
+}
+
+export class FileUploader extends Presenter {
+  state = {
+    status: 'inactive',
+    file: undefined
+  }
+
+  queue = new Subject()
+
+  render() {
+    let { status, file } = this.state
+
+    return (
+      <ActionForm
+        action={uploadFile}
+        serializer={asFormData}
+        queue={this.queue}
+        onSend={this.trackProgress}
+      >
+        <Progress status={status} file={file} onCancel={this.queue.clear} />
+
```

Comment:
New here: you can pass a `Subject` to `ActionForm` and `ActionButton`. This lets you control the underlying scheduling.

New here: `Subject:clear()`. This lets you cancel all subscriptions but keep the subject open. It's sort of like a reset.